etna/prediction_intervals/base.py

```diff
@@ -0,0 +1,190 @@
+        ts = pipeline.ts
+        self.pipeline = pipeline
+        super().__init__(pipeline.horizon)
+        self.pipeline.ts = ts
+
+    def fit(self, ts: TSDataset, save_ts: bool = True) -> "BasePredictionIntervals":
+        """Fit the pipeline or ensemble of pipelines.
+
+        Fit and apply given transforms to the data, then fit the model on the transformed data.
+
+        Parameters
+        ----------
+        ts:
+            Dataset with timeseries data.
+        save_ts:
+            Whether to save ``ts`` in the pipeline during ``fit``.
+
+        Returns
+        -------
+        :
+            Fitted instance.
+        """
+        self.pipeline.fit(ts=ts, save_ts=save_ts)
+        return self
+
+    @property
+    def ts(self) -> Optional[TSDataset]:
+        """Access internal pipeline dataset."""
+        return self.pipeline.ts
+
+    @ts.setter
+    def ts(self, ts_: Optional[TSDataset]):
+        """Set internal pipeline dataset."""
+        self.pipeline.ts = ts_
+
+    def _predict(
+        self,
+        ts: TSDataset,
+        start_timestamp: Optional[pd.Timestamp],
+        end_timestamp: Optional[pd.Timestamp],
+        prediction_interval: bool,
+        quantiles: Sequence[float],
+        return_components: bool,
+    ) -> TSDataset:
+        """Make in-sample predictions on dataset in a given range.
+
+        This method is not implemented by default. A custom implementation could be added by overriding if needed.
+
+        Parameters
+        ----------
+        ts:
+            Dataset to make predictions on.
+        start_timestamp:
+            First timestamp of prediction range to return, should be >= than first timestamp in ``ts``;
+            expected that beginning of each segment <= ``start_timestamp``;
+            if isn't set the first timestamp where each segment began is taken.
+        end_timestamp:
+            Last timestamp of prediction range to return; if isn't set the last timestamp of ``ts`` is taken.
+            Expected that value is less or equal to the last timestamp in ``ts``.
+        prediction_interval:
+            If ``True`` returns prediction interval.
+        quantiles:
+            Levels of prediction distribution. By default 2.5% and 97.5% taken to form a 95% prediction interval.
+        return_components:
+            If ``True`` additionally returns forecast components.
+
+        Returns
+        -------
+        :
+            Dataset with predictions in ``[start_timestamp, end_timestamp]`` range.
+        """
+        raise NotImplementedError(
+            "In-sample sample prediction is not supported! See documentation on how it could be implemented."
+        )
+
+    def _forecast(self, ts: TSDataset, return_components: bool) -> TSDataset:
+        """Make point forecasts using base pipeline or ensemble."""
+        return self.pipeline._forecast(ts, return_components)
+
+    def save(self, path: pathlib.Path):
```

Comment:
May be it is better to make them abstract? May you clarify the usage of these methods in the future